test/sequential/xyk-pallet.api.maintenance-config.test.ts

```diff
@@ -0,0 +1,350 @@
+    Sudo.sudoAs(
+      sudo,
+      Xyk.createPool(
+        MGA_ASSET_ID,
+        defaultPoolVolumeValue,
+        firstCurrency,
+        defaultPoolVolumeValue
+      )
+    ),
+    Sudo.sudoAsWithAddressString(
+      foundationAccountAddress,
+      Maintenance.switchMaintenanceModeOff()
+    )
+  );
+
+  liqId = await getLiquidityAssetId(MGA_ASSET_ID, firstCurrency);
+
+  await Sudo.batchAsSudoFinalized(
+    Assets.promotePool(liqId.toNumber(), 20),
+    Sudo.sudoAs(
+      testUser1,
+      Xyk.mintLiquidity(MGA_ASSET_ID, firstCurrency, defaultPoolVolumeValue)
+    )
+  );
+});
+
+test("maintenance- try to change Maintenance Mode with a non-foundation account THEN it failed", async () => {
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      foundationAccountAddress,
+      Maintenance.switchMaintenanceModeOn()
+    )
+  );
+
+  const switchMaintenanceModeByUser = await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      testUser1.keyRingPair.address,
+      Maintenance.switchMaintenanceModeOff()
+    )
+  );
+
+  await getSudoError(switchMaintenanceModeByUser, "NotFoundationAccount");
+
+  const switchMaintenanceModeBySudo = await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      sudo.keyRingPair.address,
+      Maintenance.switchMaintenanceModeOff()
+    )
+  );
+
+  await getSudoError(switchMaintenanceModeBySudo, "NotFoundationAccount");
+
+  const switchMaintenanceModeByFoundation = await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      foundationAccountAddress,
+      Maintenance.switchMaintenanceModeOff()
+    )
+  );
+
+  const filteredEvent = switchMaintenanceModeByFoundation.filter(
+    (extrinsicResult) => extrinsicResult.method === "SudoAsDone"
+  );
+  const eventIndex = JSON.parse(JSON.stringify(filteredEvent[0].event.data[0]));
+  expect(eventIndex.ok).toBeDefined();
+});
+
+test("check UpgradabilityOn can only be set after MaintenanceModeOn is set and MaintenanceMode works without UpgradabilityOn", async () => {
+  await checkMaintenanceStatus(false, false);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      foundationAccountAddress,
+      Maintenance.switchUpgradabilityInMaintenanceModeOn()
+    )
+  );
+
+  await checkMaintenanceStatus(false, false);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      foundationAccountAddress,
+      Maintenance.switchMaintenanceModeOn()
+    )
+  );
+
+  await checkMaintenanceStatus(true, false);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      foundationAccountAddress,
+      Maintenance.switchUpgradabilityInMaintenanceModeOn()
+    )
+  );
+
+  await checkMaintenanceStatus(true, true);
+
+  await Sudo.batchAsSudoFinalized(
+    Sudo.sudoAsWithAddressString(
+      foundationAccountAddress,
+      Maintenance.switchMaintenanceModeOff()
+    )
+  );
+
+  await checkMaintenanceStatus(false, false);
```

Comment:
does this test pass? perhaps you discovered a bug!
I think that first we need to disable Upgradability and then maintenanceMode

Reply:
Upgradability doesn't work without MaintenanceMode, that's why when you deactivate MaintenanceMode, you also deactivate Upgradability

Reply:
you are right, it works like that :O